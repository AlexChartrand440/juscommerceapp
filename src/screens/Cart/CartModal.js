import React from 'react';
import { Modal, Text } from 'react-native';
import styled from 'styled-components';
import { Appbar } from 'react-native-paper';

import CartModalItem from '../../components/CartModalItem';
import Button from '../../components/Button';
import Empty from '../../components/Empty';
import NavigationService from '../../NavigationService';

const AppbarHeader = styled(Appbar.Header)`
  background-color: #ff685f;
  color: #fff;
`;

const Content = styled.View`
  flex: 1;
  background-color: #f5f5f7;
  padding-horizontal: 16;
  padding-vertical: 16;
  border-radius: 5;
`;

const Footer = styled.View`
  flex: 1;
  padding-horizontal: 16;
  align-items: center;
`;

const onPress = setVisible => {
  NavigationService.navigate('Checkout');
  setVisible(false);
};

const CartModal = ({ cart }) => {
  return (
    <Modal
      visible={cart.visible}
      presentationStyle={'fullScreen'}
      transparent={false}
    >
      <AppbarHeader>
        <Appbar.Content
          title='Carrinho de compras'
          titleStyle={{ color: '#FFF', fontSize: 16 }}
        />
        <Appbar.Action
          icon='clear'
          color='#fff'
          size={36}
          onPress={() => cart.setVisible(false)}
        />
      </AppbarHeader>
      <Content>
        {cart.list.length > 0 ? (
          cart.list.map(el => {
            return (
              <CartModalItem
                product={el.product}
                key={el.product.id}
                quantity={el.amount}
                onAdd={() => cart.add(el.product)}
                onRemove={() => cart.remove(el.product)}
                change={value => cart.change(el.product, value)}
              />
            );
          })
        ) : (
          <Empty>Carrinho vazio</Empty>
        )}
        <Footer>
          {cart.list.length > 0 && (
            <Button
              label='Concluir compra'
              onPress={() => onPress(cart.setVisible)}
            />
          )}
        </Footer>
      </Content>
    </Modal>
  );
};

export default CartModal;
