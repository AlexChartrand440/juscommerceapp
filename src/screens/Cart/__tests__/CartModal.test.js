import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import CartModal from '../CartModal';

const emptyCart = {
  list: [],
  setList: jest.fn()
};

const oneItemCart = {
  list: [
    {
      product: {
        id: 1,
        name: 'test product1',
        description: 'test description 1',
        image:
          'https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681',
        price: 123
      },
      amount: 10
    }
  ],
  setList: jest.fn()
};

test('test empty cart', () => {
  const wrapper = renderer.create(<CartModal cart={emptyCart} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

test('test one item cart', () => {
  const wrapper = renderer.create(<CartModal cart={oneItemCart} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
