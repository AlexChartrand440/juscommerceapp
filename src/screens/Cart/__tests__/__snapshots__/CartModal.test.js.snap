// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test empty cart 1`] = `
<Modal
  hardwareAccelerated={false}
  presentationStyle="fullScreen"
  transparent={false}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#ff685f",
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff685f",
          "color": "#fff",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                },
              ],
            ]
          }
        >
          Carrinho de compras
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessible={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 18,
                "height": 36,
                "justifyContent": "center",
                "margin": 6,
                "overflow": "visible",
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 36,
                  "width": 36,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 36,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f7",
          "borderRadius": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 16,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ff685f",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#444444",
              "fontSize": 18,
              "fontWeight": "300",
            },
          ]
        }
      >
        Adicione itens no seu carrinho!
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingHorizontal": 16,
          },
        ]
      }
    />
  </View>
</Modal>
`;

exports[`test one item cart 1`] = `
<Modal
  hardwareAccelerated={false}
  presentationStyle="fullScreen"
  transparent={false}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#ff685f",
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff685f",
          "color": "#fff",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "fontSize": 17,
                },
                Object {
                  "color": "#FFF",
                  "fontSize": 16,
                },
              ],
            ]
          }
        >
          Carrinho de compras
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessible={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 18,
                "height": 36,
                "justifyContent": "center",
                "margin": 6,
                "overflow": "visible",
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 36,
                  "width": 36,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 36,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f7",
          "borderRadius": 5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "marginBottom": 12,
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          },
        ]
      }
      white={true}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 5,
                  "height": 80,
                  "width": 100,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingLeft": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#202020",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
              ]
            }
          >
            test product1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#444444",
                  "fontSize": 12,
                  "fontWeight": "200",
                },
              ]
            }
          >
            test description 1
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ff685f",
                    "flexBasis": 0,
                    "flexGrow": 2,
                    "flexShrink": 1,
                    "fontSize": 14,
                    "fontWeight": "500",
                  },
                ]
              }
            >
              R$ 123
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "flex-end",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 2,
                    "flexShrink": 1,
                    "justifyContent": "space-around",
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "height": 15,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff685f",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgb(231, 231, 231)",
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    Object {
                      "backgroundColor": "#fff",
                      "flex": 2,
                      "fontSize": 12,
                      "height": 50,
                      "textAlign": "right",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ff685f",
                      "bottom": 0,
                      "height": 2,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": Array [
                        Object {
                          "scaleY": 0.5,
                        },
                      ],
                    }
                  }
                />
                <TextInput
                  allowFontScaling={true}
                  editable={true}
                  keyboardType="numeric"
                  maxLength={2}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  selectionColor="#ff685f"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "fontSize": 16,
                        "margin": 0,
                        "minHeight": 58,
                        "paddingHorizontal": 12,
                        "textAlign": "left",
                        "zIndex": 1,
                      },
                      Object {
                        "paddingVertical": 15,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "Helvetica Neue",
                        "textAlignVertical": "center",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value="10"
                />
              </View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "height": 15,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff685f",
                        "fontSize": 15,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingHorizontal": 16,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#ff685f",
            "borderColor": "transparent",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "height": 45,
            "justifyContent": "center",
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 0,
            "width": 250,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Helvetica Neue",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "HelveticaNeue-Medium",
                    },
                    Object {
                      "fontFamily": "HelveticaNeue-Medium",
                    },
                  ],
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                      "fontWeight": "300",
                      "width": "100%",
                    },
                  ]
                }
              >
                Concluir compra
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
